p2p/bootnode.go

```diff
@@ -61,12 +64,59 @@ func NewUDPBootnodes(ctx context.Context, config Config, peers []Peer,
 				// Do not include ourselves as bootnode.
 				continue
 			}
-			node := p.Enode // Copy loop variable
-			resp = append(resp, &node)
+
+			resp = append(resp, NewMutablePeer(p))
+		}
+	}
+
+	if len(resp) == 0 {
+		return nil, nil
+	}
+
+	// Wait until at least one bootnode ENR is resolved
+	ctx, cancel := context.WithTimeout(ctx, time.Minute)
+	defer cancel()
+	for ctx.Err() == nil {
+		var resolved bool
+		for _, node := range resp {
+			if _, ok := node.Peer(); ok {
+				resolved = true
+			}
 		}
+		if resolved {
+			return resp, nil
+		}
+		time.Sleep(time.Second * 1)
```

Comment:
discussion: why sleep for 1s?

Reply:
checking for a `resolved` bootnode 60 times before returning?..

Reply:
one should always avoid spinning or thrashing, so backoff is good.